src/utils/appLogic.ts

```diff
@@ -8,6 +9,14 @@ export const isDeviceInPortrait = (screen: ScaledSize) =>
 
 export const areFilmsFetched = (films: Film[]) => films && films.length !== 0;
 
+export const getFilmFromId = (films: Film[], filmId: string): FilmDetails => {
+  const film = films.find(filmMatch => filmMatch.id === filmId);
+  if (!film) {
+    throw new Error(`Film with ID ${filmId} not found.`);
+  }
+  return film;
```

Comment:
You are missing non-optional properties here, return them or mark them as optional.